Source/Laser/LaserProfilesImpl/LaserProfileFromFile.cpp

```diff
@@ -0,0 +1,570 @@
+    if(t >= m_params.t_max)
+        return;
+    const auto idx_times = find_left_right_time_indices(t);
+    const auto idx_t_left = idx_times.first;
+    const auto idx_t_right = idx_times.second;
+    //Load data chunk if needed
+    if(idx_t_right >  m_params.last_time_index){
+        if (m_params.file_in_lasy_format){
+        read_data_t_chunk(idx_t_left, idx_t_left+m_params.time_chunk_size);
+        } else{
+        read_binary_data_t_chunk(idx_t_left, idx_t_left+m_params.time_chunk_size);
+        }
+    }
+}
+
+void
+WarpXLaserProfiles::FromFileLaserProfile::fill_amplitude (
+    const int np,
+    Real const * AMREX_RESTRICT const Xp, Real const * AMREX_RESTRICT const Yp,
+    Real t, Real * AMREX_RESTRICT const amplitude) const
+{
+    t += m_params.t_min - m_params.t_delay;
+    //Amplitude is 0 if time is out of range
+    if(t < m_params.t_min ||  t > m_params.t_max){
+        amrex::ParallelFor(np,
+            [=] AMREX_GPU_DEVICE (int i) {
+                amplitude[i] = 0.0_rt;});
+        return;
+    }
+    //Find left time index
+    const auto idx_t_left = find_left_right_time_indices(t).first;
+    if(idx_t_left <  m_params.first_time_index){
+        Abort("Something bad has happened with the simulation time");
+    }
+    if (m_params.file_in_lasy_format){
+        internal_fill_amplitude_uniform(idx_t_left, np, Xp, Yp, t, amplitude);
+    } else{
+        internal_fill_amplitude_uniform_binary(idx_t_left, np, Xp, Yp, t, amplitude);
+    }
+}
+
+void
+WarpXLaserProfiles::FromFileLaserProfile::parse_lasy_file(std::string lasy_file_name)
+{
+#ifdef WARPX_USE_OPENPMD
+    if(ParallelDescriptor::IOProcessor()){
+        auto series = io::Series(lasy_file_name, io::Access::READ_ONLY);
+        auto i = series.iterations[0];
+        auto E = i.meshes["laserEnvelope"];
+
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(E.getAttribute("dataOrder").get<std::string>() == "C",
+                                         "Reading from files with non-C dataOrder is not implemented");
+        auto axisLabels = E.getAttribute("axisLabels").get<std::vector<std::string>>();
+        auto fileGeom = E.getAttribute("geometry").get<std::string>();
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(fileGeom == "cartesian", "WarpX can only read laser files with cartesian 3D geometry.");
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(axisLabels[0] == "t" && axisLabels[1] == "y" && axisLabels[2] == "x",
+                                         "WarpX expects laser files with axisLabels {t, y, x}.");
+
+        auto E_laser = E[io::RecordComponent::SCALAR];
+        auto extent = E_laser.getExtent();
+        //Dimensions of lasy file datas: {t,y,x}
+        m_params.nt = extent[0];
+        m_params.ny = extent[1];
+        m_params.nx = extent[2];
+        if(m_params.nt <= 1) Abort("nt in lasy file must be >=2");
+        if(m_params.nx <= 1) Abort("nx in lasy file must be >=2");
+        if(m_params.ny <= 1) Abort("ny in lasy file must be >=2 in 3D");
```

Comment:
I would suggest this change
```suggestion
        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(m_params.nt > 1, "nt in lasy file must be >=2");
        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(m_params.nx > 1, "nx in lasy file must be >=2");
        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(m_params.ny > 1, "ny in lasy file must be >=2 in 3D");
```